integration/tests/cook/util.py

```diff
@@ -43,46 +55,114 @@ class User(object):
 
     def __init__(self, name):
         self.name = name
+
+    def __enter__(self):
+        logger.debug(f'Switching to user {self.name}')
+
+    def __exit__(self, ex_type, ex_val, ex_trace):
+        logger.debug(f'Switching back from user {self.name}')
+
+
+class _BasicAuthUser(_AuthenticatedUser):
+    """
+    Object representing a Cook user with HTTP Basic Auth credentials.
+    """
+
+    def __init__(self, name):
+        super().__init__(name)
         self.auth = (name, '')
         self.previous_auth = None
 
     def __enter__(self):
         global session
-        logger.debug(f"Switching to user {self.name}")
+        super().__enter__()
         assert self.previous_auth is None
         self.previous_auth = session.auth
         session.auth = self.auth
 
     def __exit__(self, ex_type, ex_val, ex_trace):
         global session
-        logger.debug(f"Switching back from user {self.name}")
+        super().__exit__(ex_type, ex_val, ex_trace)
         assert self.previous_auth is not None
         session.auth = self.previous_auth
         self.previous_auth = None
 
 
+class _KerberosUser(_AuthenticatedUser):
+    """
+    Object representing a Cook user with Kerberos credentials.
+    """
+
+    def __init__(self, name):
+        super().__init__(name)
+        subcommand = f'{_kerberos_auth_cmd} {name}'
```

Comment:
Do you mean to append the username here? I don't think that's in the documentation

Reply:
You're right. I'll delete that.